state/remote/http.go

```diff
@@ -44,20 +44,40 @@ func httpFactory(conf map[string]string) (Client, error) {
 		}
 	}
 
-	return &HTTPClient{
+	ret := &HTTPClient{
 		URL:    url,
 		Client: client,
-	}, nil
+	}
+	if username, ok := conf["username"]; ok && username != "" {
+		ret.Username = username
+	}
+	if password, ok := conf["password"]; ok && password != "" {
+		ret.Password = password
+	}
+	return ret, nil
 }
 
 // HTTPClient is a remote client that stores data in Consul or HTTP REST.
 type HTTPClient struct {
-	URL    *url.URL
-	Client *http.Client
+	URL      *url.URL
+	Client   *http.Client
+	Username string
+	Password string
 }
 
 func (c *HTTPClient) Get() (*Payload, error) {
-	resp, err := c.Client.Get(c.URL.String())
+	req, err := http.NewRequest("GET", c.URL.String(), nil)
+	if err != nil {
+		return nil, err
+	}
+
+	// Prepare the request
+	if c.Username != "" {
```

Comment:
Unnecesary. The empty string is the zero value, so in this case the HTTP client will just use an empty password, which is totally legit for HTTP basic auth.